### PR TITLE
Reopening the Closed Pull Request

### PR DESCRIPTION
**Pull request by kunal-pmj**
@vinpai  Added two samples Prashant Altekar is OOO till 25th June

ContentLibraryDelete.java : Takes datastore name as mandatory parameter and deletes all the Content Libraries manages by the current VC in this Datastore. An optional parameter content Library name can also be passed to delete a Content Library by Name.
PowerLifeCycle.java : Demonstrates the Power Life Cycle of a VM. id cleardata is passed the VM is powered off after performing ON, Suspend, Resume

### CODE REVIEW

File: src/main/java/vmware/samples/contentlibrary/crud/LibraryCrud.java [hunk 1]
```diff
@@ -113,13 +111,36 @@ protected void run() throws Exception {
...
     }
+    /**
+     * Creates a datastore storage backing.
+     *
+     * @return the storage backing
+     */
+    private StorageBacking createStorageBacking() {
+    	
+        // Retrieve the MoRef of a VC datastore using VIM APIs
+        ManagedObjectReference dsMoref = VimUtil.getEntityByName(
```
Comment (vinpai) [inline src/main/java/vmware/samples/contentlibrary/crud/LibraryCrud.java]: Why can't we use the Datastore enumeration API here ?

File: src/main/java/vmware/samples/contentlibrary/delete/ContentLibraryDelete.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+import com.vmware.content.LibraryModel;
+import com.vmware.content.LibraryTypes;
+import com.vmware.content.library.StorageBacking;
+import com.vmware.vim25.ManagedObjectReference;
+
+import vmware.samples.common.SamplesAbstractBase;
+import vmware.samples.common.vim.helpers.VimUtil;
+import vmware.samples.contentlibrary.client.ClsApiClient;
+
+public class ContentLibraryDelete extends SamplesAbstractBase {
```
Comment (vinpai) [inline src/main/java/vmware/samples/contentlibrary/delete/ContentLibraryDelete.java]: Why do we need a separate sample for delete ? Shouldn't this be in LibraryCrud ?

File: src/main/java/vmware/samples/vcenter/vm/power/PowerLifeCycle.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            .argName("VM NAME")
+            .build();
+        List<Option> optionList = Collections.singletonList(vmNameOption);
+        super.parseArgs(optionList, args);
+        this.vmName =  (String) parsedOptions.get("vmname");        
+    }
+
+    @Override
+    protected void setup() throws Exception {
+        vmId = VmHelper.getVM(vapiAuthHelper.getStubFactory(),
```
Comment (vinpai) [inline src/main/java/vmware/samples/vcenter/vm/power/PowerLifeCycle.java]: Please refer all instance variables with "this." for consistency with other samples.

File: src/main/java/vmware/samples/vcenter/vm/power/PowerLifeCycle.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                sessionStubConfig,
+                vmName);
+        System.out.println("Using VM: " + vmName + " (vmId="
+            + this.vmId + " ) for Power Operations sample.");
+        vmPowerService = vapiAuthHelper.getStubFactory().createStub(
```
Comment (vinpai) [inline src/main/java/vmware/samples/vcenter/vm/power/PowerLifeCycle.java]: this.vmPowerService

File: src/main/java/vmware/samples/contentlibrary/delete/ContentLibraryDelete.java (hunk 2 above)
Comment (kunal-pmj) [reply src/main/java/vmware/samples/contentlibrary/delete/ContentLibraryDelete.java]: LibraryCRUD is not the only sample creating Content Libraries. This sample was added to all the content libraries in the DC or one library by name.

File: src/main/java/vmware/samples/vcenter/vm/power/PowerLifeCycle.java (hunk 3 above)
Comment (kunal-pmj) [reply src/main/java/vmware/samples/vcenter/vm/power/PowerLifeCycle.java]: Done!!

File: src/main/java/vmware/samples/vcenter/vm/power/PowerLifeCycle.java (hunk 4 above)
Comment (kunal-pmj) [reply src/main/java/vmware/samples/vcenter/vm/power/PowerLifeCycle.java]: Done

**Pull request merged**